### PR TITLE
Set & delete users template

### PR DESCRIPTION
**Pull request by mirceaulinic**
<!-- Make sure you have read http://napalm.readthedocs.io/en/latest/contributing/index.html --!>

### CODE REVIEW

**Comment by itdependsnetworks** [comment]
At least when using ansible's implementation of jinja, it will fail if the key doesn't exist. I would suggest the following: 

```
{%- for user_name, user_details in users.items() %}
{%- if user_details.get('password') %}
username {{ user_name }} secret 5 {{ user_details.password }}
{%- else %}
username {{ user_name }} nopassword
{%- endif %}
{%- if user_details.get('level') %}
username {{ user_name }} privilege {{ user_details.level }}
{%- endif %}
{%- if user_details.get('sshkeys') %}
{%- for sshkey in user_details.sshkeys %}
username {{ user_name }} sshkey {{ sshkey }}
{%- endfor %}
{%- endif %}
{%- endfor %}
```

**Comment by dbarrosop** [comment]
I agree with @itdependsnetworks. We should check the variables are present before trying to configure them. Newest EOS release support users without password, only with keys. And there might be people using passwords only instead of keys.

File: napalm_eos/templates/delete_users.j2 [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{%- for user_name, user_details in users.iteritems() %}
```
Comment (dbarrosop) [inline napalm_eos/templates/delete_users.j2]: Isn't it enough to do something like:

```
{%- for user_name, user_details in users_to_delete %}
default username {{ user_name}}
{%- endfor %}
```

**Comment by mirceaulinic** [comment]
> We should check the variables are present before trying to configure them. Newest EOS release support users without password, only with keys. And there might be people using passwords only instead of keys.

Can't agree more - that's why I have:

```
{%- if user_details.sshkeys %}
```

Which works in pure python.. but it seems to fail in other obscure implementations.

But I agree on the additional level of safety.

File: napalm_eos/templates/delete_users.j2 (hunk 1 above)
Comment (mirceaulinic) [reply napalm_eos/templates/delete_users.j2]: What does:

```
default username {{ user_name}}
```

do more specifically?

**Review by dbarrosop — APPROVED** [review]

**Pull request merged**